File: 1. TwoSum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    let len = nums.length;
    if (len < 2) return null;
    let diffs = {};
    for (let i = 0; i < len; i++) {
        if (diffs[nums[i]] !== undefined) return [diffs[nums[i]], i]
        diffs[target - nums[i]] = i; 
    }
    return null;
};

// Loop through nums
// save index of each loop with target - nums[i] as key
// That way when whenever diff[i] is not undefined, it means a previous
// number which sums up to target with diff[i] was already found.

File: 1. TwoSum/twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        if len(nums) < 2: return None
        diffs = {}
        x = 0
        while x < len(nums):
            if nums[x] in diffs: return [diffs[nums[x]], x] 
            diffs[target - nums[x]] = x 
            x+=1
        return None
